app/api/v1/predict_construction_time/app.py

```diff
@@ -0,0 +1,51 @@
+from flask import request
+from flask_restplus import Resource
+from ..api import ns
+
+process_route = "/predict_construction_time"
+
+
+@ns.route(process_route)
+class PredictModelVersions(Resource):
+    def get(self):
+        """
+        Return the list of all applicable model versions, optionally with their creation dates?
+        :return: Map<model_version_label, date_created>
+        """
+        #  DO WORK HERE
+        pass
+
+
+@ns.route(process_route + '/<int:model_ref>/cnc_quick_materials')
```

Comment:
There is very likely a 'prefix' parameter to the `route` decorator.

Reply:
but if we were shooting for the convention

`predict/v1/modelN/api`

`predict` and `v1` would get added later to the mix.

Reply:
I think the bit I'm struggling with is trying to compose the Namespace elements at multiple layers - we want the top-level {predict}, the per-version {v1}, and then the parameterized {model#} and api will probably drop in together. I _may_ have a way to deliver on that, but it doesn't feel terribly pretty.
I'll take a stab at getting the concept in my head implemented (not prettily), and then clean it up from there.